App.js

```diff
@@ -27,54 +24,127 @@ export default class App extends Component {
      * @type {{timeStamp: string, altitude: number, latitude: number, errorMessage: string, accuracy: number, speed: number, longitude: number}}
      */
     this.state = {
-      latitude: 0,
-      longitude: 0,
-      errorMessage: 'No location data',
-      speed: 0,
-      accuracy: 0,
-      altitude: 0,
-      timestamp: '',
+      coords: {},
     };
+
+    this.COORDS_ENDPOINT =
+      'https://us-central1-test-947g5.cloudfunctions.net/coords';
+    this.CONFIGS = 'https://us-central1-test-947g5.cloudfunctions.net/configs';
+    this.deviceId = DeviceInfo.getUniqueId();
   }
 
+  /**
+   * Get geolocation configurations
+   * @returns {Promise<void>}
+   */
+  getGeolocationConfigFrequency = async () => {
+    let result = await fetch(this.CONFIGS + '/geoConfig');
+    let resultJson = await result.json();
+    const DEFAULT_FREQUENCY = 30000;
+    const geolocationFrequency = resultJson.geolocationFrequency;
+    const geolocationFrequencyType = resultJson.geolocationFrequencyType;
+    return geolocationFrequencyType[0] === 'seconds'
```

Comment:
We should refactor this in future to have only the smallest unit to avoid unwanted complexity. We can derive the other units from the smallest one.